module_utils/network_lsr/argument_validator.py

```diff
@@ -353,16 +353,16 @@ def _validate_impl(self, value, name):
                 raise ValidationError(e.name, e.error_message)
             result[setting] = validated_value
         for (setting, validator) in self.nested.items():
-            if setting in seen_keys or isinstance(validator, ArgValidatorDeprecated):
+            if setting in seen_keys:
+                continue
+            if isinstance(validator, ArgValidatorDeprecated):
                 continue
             if validator.required:
                 raise ValidationError(name, "missing required key '%s'" % (setting))
-            default_value = validator.get_default_value()
-            if (
-                not self.all_missing_during_validate
-                and default_value is not ArgValidator.MISSING
-            ):
-                result[setting] = default_value
+            if not self.all_missing_during_validate:
```

Comment:
Since this changes the logic, is there a bug that is fixed by this? Why does this need to be changed?

Reply:
it slightly changes the logic. It now avoids calling `validator.get_default_value()`, when we already know that it's not used (`self.all_missing_during_validate`). It's unlikely that either has a visible effect.

It still feels more correct to only get the information, if we also are going to use it. The logic is also how someone would think. I mean, I would think "the default value is only requested `if not self.all_missing_during_validate`". That is easier to understand than "the default value is always requested, but only used if `if not self.all_missing_during_validate`.

Reply:
ok, thanks